dart/lib/src/sentry_envelope_item.dart

```diff
@@ -20,6 +21,24 @@ class SentryEnvelopeItem {
     return SentryEnvelopeItem(header, () async => attachment.content);
   }
 
+  factory SentryEnvelopeItem.fromUserFeedback(UserFeedback feedback) {
+    final cachedItem = _CachedItem(() async {
+      final jsonEncoded = jsonEncode(feedback.toJson());
+      return utf8.encode(jsonEncoded);
+    });
+
+    final getLength = () async {
+      return (await cachedItem.getData()).length;
+    };
```

Comment:
part of this block is duplicated across methods, maybe we extract to a private method

Reply:
I think we should do some refactoring in a different PR.
Currently, we try to stream the envelope's payload, but that's not really possible because we need to know the size of the payload for the envelope header. We should instead stream the whole envelope. That way we could get rid of this kinda weird code alltogether.